apps/wallets/smart-wallet/next/src/app/template.tsx

```diff
@@ -1,17 +1,44 @@
+"use client";
+
 import { Header } from "@/components/header";
 import { Toaster } from "@/components/toaster";
+import { env } from "@/env";
+import { useEffect, useState } from "react";
+
+import { CrossmintAuthProvider } from "@crossmint/client-sdk-react-ui";
 
 // Reference: https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts#templates
 // Since our home page is located in the root directory, we cannot use a layout component to wrap it.
 // This is because the root layout in nextjs is server-side only, and a couple of components are client-side (see below).
 // Therefore, we are using a template component to handle client-side components and behaviors.
 
 export default function Template({ children }: { children: React.ReactNode }) {
-    return (
-        <div>
+    const [isMounted, setIsMounted] = useState(false);
+    useEffect(() => {
+        setIsMounted(true);
+    }, []);
+
+    // This is a workaround to prevent a hydration error when using the CrossmintAuthProvider.
+    // In theory, this auth provider should be added to the providers.tsx file, but due to that file running initially server-side, it
+    // causes a hydration error when the auth modal is opened for whatever reason.
+    // Checking for mount here also prevents the auth provider from throwing a "Smart Wallet SDK should only be used client side." error.
+    return !isMounted ? (
+        <>
             <Header />
             <Toaster />
             {children}
-        </div>
+        </>
+    ) : (
```

Comment:
Seems weird that this isn't already the default behavior XD

Reply:
Is this is the case when you do a production build too?

Reply:
next is sometimes weird in dev mode

Reply:
Nextjs is definitely weird in dev mode. iirc building would throw because of that error.